neo/Prompt/Commands/Send.py

```diff
@@ -14,81 +15,165 @@
 import traceback
 
 
-def construct_and_send(prompter, wallet, arguments, prompt_password=True):
-    try:
-        if not wallet:
-            print("please open a wallet")
-            return False
-        if len(arguments) < 3:
-            print("Not enough arguments")
+def construct_send_basic(prompter, wallet, arguments):
+    if not wallet:
+        print("please open a wallet")
+        return False
+    if len(arguments) < 3:
+        print("Not enough arguments")
+        return False
+
+    arguments, from_address = get_from_addr(arguments)
+    arguments, priority_fee = get_fee(arguments)
+    arguments, user_tx_attributes = get_tx_attr_from_args(arguments)
+    arguments, owners = get_owners_from_params(arguments)
+    to_send = get_arg(arguments)
+    address_to = get_arg(arguments, 1)
+    amount = get_arg(arguments, 2)
+
+    assetId = get_asset_id(wallet, to_send)
+    if assetId is None:
+        print("Asset id not found")
+        return False
+
+    scripthash_to = lookup_addr_str(wallet, address_to)
+    if scripthash_to is None:
+        print("invalid address")
+        return False
+
+    scripthash_from = None
+    if from_address is not None:
+        scripthash_from = lookup_addr_str(wallet, from_address)
+        if scripthash_from is None:
+            print("invalid address")
             return False
 
-        arguments, from_address = get_from_addr(arguments)
-        arguments, user_tx_attributes = get_tx_attr_from_args(arguments)
-        arguments, owners = get_owners_from_params(arguments)
-        arguments, priority_fee = get_fee(arguments)
-        to_send = get_arg(arguments)
-        address_to = get_arg(arguments, 1)
-        amount = get_arg(arguments, 2)
-
-        if float(amount) == 0:
-            print("amount cannot be 0")
+    # if this is a token, we will use a different
+    # transfer mechanism
+    if type(assetId) is NEP5Token:
+        return do_token_transfer(assetId, wallet, from_address, address_to, amount_from_string(assetId, amount), tx_attributes=user_tx_attributes)
+
+    f8amount = Fixed8.TryParse(amount, require_positive=True)
+    if f8amount is None:
+        print("invalid amount format")
+        return False
+    if float(str(f8amount)) == 0:
+        print("amount cannot be 0")
+        return False
+
+    if type(assetId) is UInt256 and f8amount.value % pow(10, 8 - Blockchain.Default().GetAssetState(assetId.ToBytes()).Precision) != 0:
+        print("incorrect amount precision")
+        return False
+
+    fee = Fixed8.Zero()
+    if priority_fee is not None:
+        fee = priority_fee
+        if fee < Fixed8.Zero():
+            print("fee cannot be negative")
             return False
 
+    output = TransactionOutput(AssetId=assetId, Value=f8amount, script_hash=scripthash_to)
+    contract_tx = ContractTransaction(outputs=[output])
+    return contract_tx, scripthash_from, fee, owners, user_tx_attributes
```

Comment:
the return signature of a function must be concise. Meaning you can't have success return a 5 argument tuple while failure returns a single boolean. This will exceptions when you try to use the command as follows:
 ```
contract_tx, scripthash_from, fee, owners, user_tx_attributes = construct_send_basic(prompt, wallet, bad_args)
```
`bad_args` will cause the function to returns  `False` and then you'll get an exception saying something like (from the top of my head):
```
TypeError: 'bool' object is not iterable
```
because it's expecting an tuple (which is iterable) to unpack 5 values from and instead gets a boolean